include/alpaka/warp/WarpUniformCudaHipBuiltIn.hpp

```diff
@@ -142,6 +150,12 @@ namespace alpaka
                 __device__ static auto ballot(
                     warp::WarpUniformCudaHipBuiltIn const & warp,
                     std::int32_t predicate)
+                    // return type is required by the compiler
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+                    -> unsigned
+#else
+                    -> uint64_t
+#endif
```

Comment:
May we want to guard those with some `#if BOOST_COMP_NVCC`?
This would also document the reason whythose are necessary.

Reply:
IMO we can keep it for clang too, in #1092 it will than use always fixed size types without any side effect. This will guarantee that fixed size types always work.